Webapp/sources/blueprints/polymer_novel_compound/routes.py

```diff
@@ -0,0 +1,44 @@
+from flask import Response, jsonify, request
+from flask_login import login_required
+from sources import services
+from sources.auxiliary import abort_if_user_not_in_workbook
+
+from . import polymer_novel_compound_bp
+
+
+@polymer_novel_compound_bp.route("/_polymer_novel_compound", methods=["GET", "POST"])
+@login_required
+def polymer_novel_compound() -> Response:
+    """
+    Adds a novel compound to the database and links it to the workbook of the current reaction.
+    The novel compound data is saved from the data the user enters in the form.
+    This function is triggered when a novel structure is drawn in the sketcher.
+
+    Returns:
+        Flask Response with feedback about the operation.
+    """
+    # get the active workbook and verify user belongs
+    workgroup_name = str(request.form.get("workgroup"))
+    workbook_name = str(request.form.get("workbook"))
+    workbook = services.workbook.get_workbook_from_group_book_name_combination(
+        workgroup_name, workbook_name
+    )
+    abort_if_user_not_in_workbook(workgroup_name, workbook_name, workbook)
```

Comment:
@JoeHeeley2 - could a decorator be used here instead?

Reply:
yeah you could use the @workbook_member_required decorator. Would have to ensure that workbook and workgroup are passed as kwargs to the function (eg/ def polymer_novel_compound(workgroup, workbook)) then the decorator would handle the request.form.get and pass them to the function